pymc3/distributions/timeseries.py

```diff
@@ -176,6 +184,84 @@ def logp(self, value):
         return at.sum(innov_like) + at.sum(init_like)
 
 
+class ARMARV(RandomVariable):
+    name = "ARMA"
+    ndim_supp = 0
+    ndims_params = [1, 1, 0, 0]
+    dtype = "floatX"
+    _print_name = ("ARMA", "\\operatorname{ARMA}")
+
+    def __call__(self, phi=[], theta=[], loc=[0.0], scale=[1.0], size=None, **kwargs) -> TensorVariable:
+        return super().__call__(phi, theta, loc, scale, size=size, **kwargs)
+
+    @classmethod
+    def rng_fn(
+            cls,
+            rng: np.random.default_rng(),
+            phi: np.ndarray,
+            theta: np.ndarray,
+            loc: np.ndarray,
+            scale: np.ndarray,
+            size: Tuple[int, ...],
+    ) -> np.ndarray:
+
+        phi = np.r_[1, -phi]
+        theta = np.r_[1, theta]
+        if size:
+            x = rng.normal(loc=loc, scale=scale, size=size)
+        else:
+            x = rng.normal(loc=loc, scale=scale, size=100)
+        return signal.lfilter(theta, phi, x, axis=-1)
+
+
+ARMArv = ARMARV()
+
+
+class ARMA(Continuous):
+    rv_op = ARMArv
+
+    @classmethod
+    def dist(cls, P=None, Q=None, phi=None, theta=None, loc=None, scale=None, *args, **kwargs):
+        if (phi is None and P is None) or (theta is None and Q is None):
+            raise ValueError('Specify (phi or P) and (theta or Q)')
+
+        if phi is not None:
+            phi = at.as_tensor_variable(floatX(phi))
+        if phi is None:
+            phi = Uniform("phi", -0.99, 0.99, shape=P)
+        if theta is not None:
+            theta = at.as_tensor_variable(floatX(theta))
+        if theta is None:
+            theta = Uniform("theta", -0.99, 0.99, size=Q)
+        if scale is None:
+            scale = pm.HalfNormal("scale", 5.0)
+        if loc is None:
+            loc = pm.Normal("loc", 0, 10.0)
+        return super().dist([phi, theta], loc=loc, scale=scale, *args, **kwargs)
+
+    def logp(Y, phi=None, theta=None, loc=None, scale=None):
+        T = Y.shape[0]
+        P = phi.shape[0]
+        Q = theta.shape[0]
+        Y_lagged = at.zeros(shape=(T, P))
+        for i in range(P.eval()):
+            Y_lagged = at.set_subtensor(Y_lagged[(i + 1):, i], Y[:-(i + 1)])
+
+        mu = at.fill(Y, loc) + at.dot(Y_lagged, at.transpose(phi))
```

Comment:
Comments here explaining the AR part and the MA part would be helpful

Reply:
thanks, will do